### PR TITLE
Add goldens for egress rule

### PR DESCRIPTION
**Pull request by rshriram**
**What this PR does / why we need it**:
Improving test coverage for egress rules. Built on top of previous PR.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Contributes to #1232 
**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Review by GregHanson — APPROVED** [review]
lgtm

File: proxy/envoy/discovery_test.go [hunk 1]
```diff
@@ -83,6 +83,14 @@ func makeDiscoveryRequest(ds *DiscoveryService, method, url string, t *testing.T
 	return body
 }
 
+func commonSetup(t *testing.T) (*proxyconfig.MeshConfig, model.ConfigStore, *DiscoveryService) {
+	mesh := makeMeshConfig()
+	registry := memory.Make(model.IstioConfigTypes)
+	addConfig(registry, egressRule, t)
```
Comment (kyessenov) [inline proxy/envoy/discovery_test.go]: please remove egressRule from common setup. The tests should test each rule/config in isolation.

**Review by kyessenov — CHANGES_REQUESTED** [review]

**Review by kyessenov — APPROVED** [review]

**Comment by rshriram** [comment]
One at a time wins again!

**Comment by Arttii** [comment]
@rshriram So I after going from a bunch of issues I stumbled here concerning the headless service auth. I was wondering why was it disabled, I cannot find anywhere why exactly is it like this. Is it a perf problem or something?

**Comment by ldemailly** [comment]
not a perf problem

**Pull request merged**